src/Raven.Server/Web/System/AdminPrometheusMonitoringHandler.cs

```diff
@@ -0,0 +1,501 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.ServerWide;
+using Raven.Server.Commercial;
+using Raven.Server.Documents;
+using Raven.Server.Monitoring;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils.Monitoring;
+using Sparrow;
+using Sparrow.LowMemory;
+
+namespace Raven.Server.Web.System
+{
+    public sealed class AdminPrometheusMonitoringHandler : RequestHandler
+    {
+        public const string PrometheusContentType = "text/plain; version=0.0.4; charset=utf-8";
+        public const string MetricsPrefix = "ravendb_";
+
+        public class EnumHelp
+        {
+            private EnumHelp()
+            {
+            }
+
+            public static readonly string LicenseType = FormatEnumHelp<LicenseType>();
+            public static readonly string LowMemorySeverity = FormatEnumHelp<LowMemorySeverity>();
+            public static readonly string RachisState = FormatEnumHelp<RachisState>();
+            public static readonly string IndexPriority = FormatEnumHelp<IndexPriority>();
+            public static readonly string IndexState = FormatEnumHelp<IndexState>();
+            public static readonly string IndexStatus = FormatEnumHelp<IndexRunningStatus>();
+            public static readonly string IndexLockMode = FormatEnumHelp<IndexLockMode>();
+            public static readonly string IndexRunningStatus = FormatEnumHelp<IndexRunningStatus>();
+            public static readonly string IndexType = FormatEnumHelp<IndexType>();
+            
+            private static string FormatEnumHelp<TEnum>() where TEnum : struct, Enum
+            {
+                var values = Enum.GetValues<TEnum>();
+                return "Values: " + string.Join(", ", values.Select(x => ((int)(object)x) + " = " + x));
+            }
+        }
+
+        [RavenAction("/admin/monitoring/v1/prometheus", "GET", AuthorizationStatus.Operator)]
+        public async Task Metrics()
+        {
+            ServerStore.LicenseManager.AssertCanUseMonitoringEndpoints();
+
+            HttpContext.Response.ContentType = PrometheusContentType;
+
+            var skipServer = GetBoolValueQueryString("skipServerMetrics", false) ?? false;
+            var skipDatabases = GetBoolValueQueryString("skipDatabasesMetrics", false) ?? false;
+            var skipIndexes = GetBoolValueQueryString("skipIndexesMetrics", false) ?? false;
+            var skipCollections = GetBoolValueQueryString("skipCollectionsMetrics", false) ?? false;
+
+            var provider = new MetricsProvider(Server);
+
+            var databases = GetDatabases();
+
+            var responseStream = ResponseBodyStream();
```

Comment:
it is always good to wrap those things in using

Reply:
done